test/test_database.py

```diff
@@ -634,6 +636,42 @@ def test_with_options(self):
             self.assertEqual(
                 getattr(db2, opt), newopts.get(opt, getattr(db1, opt)))
 
+    @client_context.require_auth
```

Comment:
One thought I had for improving this would be splitting it up into multiple functions like `test_helpers_with_comment_require_auth` to allow us to test the operations that allow so on as many configurations as possible. I only put these requires here because there are 1 or 2 functions that I'm testing that requires them, so they could easily be split off.

Reply:
Sounds like a good use case for parameterized tests using `pytest`.  :smile:

Reply:
Which operations require auth? I can't recall any that do.

Reply:
Fixed. Removed auth requirement for database.